magicmaze.js

```diff
@@ -442,8 +444,33 @@ function cellClickHandler (dojo, game, cell, evt) {
   const possibleWarp = 'warpFn' in cell && game.abilities[game.player_id].indexOf('P') !== -1
   const possibleMoves = eligibleActions(game, cell.posX, cell.posY)
 
+  const tokenNames = {
+    0: _('Elf'),
+    1: _('Dwarf'),
+    2: _('Barbarian'),
+    3: _('Mage'),
+  }
+
+  // first reset page title
+  game.updatePageTitle()
   if (possibleMoves.length > 1) {
-    game.showMessage(_('Possibly ambiguous move, use the specific arrow to move the desired character'), 'error')
+    game.statusBar.setTitle(_('Ambiguous move. Select a token.'))
+    for (const move of possibleMoves) {
+      const [token, ability, dx, dy] = move
+      game.statusBar.addActionButton(
+        (`<span id="mm_meeple${token}"></span> ` + 
+         dojo.string.substitute(_('Move the ${token}'), {token: tokenNames[token]})),
+        () => {
+          if (ability === 'R') {
```

Comment:
does this work? I think the warp overlay populates before we even get to this code

Reply:
Ah, there are two cases: if one or more token can move onto the teleporter. If only one token can move on the teleporter it does not trigger the new logic.
And also if there is no option to warp a token with this query. I should probably add that.